src/lib/logical_query_plan/intersect_node.cpp

```diff
@@ -28,6 +28,12 @@ bool IntersectNode::is_column_nullable(const ColumnID column_id) const {
   return left_input()->is_column_nullable(column_id) || right_input()->is_column_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> IntersectNode::unique_constraints() const {
+  // Because INTERSECT acts as a pure filter for the left input table, all unique constraints from the left input node
+  // can be forwarded.
+  return _forward_left_unique_constraints();
+}
+
```

Comment:
Could it become possible for both input nodes to have differing unique constraints? If so, we could try to merge them.
However, if so, I'd add a ToDo-comment only since Hyrise does not yet support set operations.

Reply:
Yes, I believe that both sides' constraints can be forwarded.